### PR TITLE
Data Backup Feature

### PR DESCRIPTION
**Pull request by Yukinyaa**
some enhancement and discussion should be made

### CODE REVIEW

**Comment by dragonjet** [comment]
first, jshint is failing on the build :sweat_smile: 
https://travis-ci.org/KC3Kai/KC3Kai/builds/116291659

```
src/library/modules/Kcsapi.js
    788 |        "api_req_sortie/ld_airbattle":function(params, response, headers){
                                              ^ Duplicate key 'api_req_sortie/ld_airbattle'.
    814 |        "api_req_combined_battle/ld_airbattle":function(params, response, headers){
                                                       ^ Duplicate key 'api_req_combined_battle/ld_airbattle'.
```

**Comment by Yukinyaa** [comment]
@dragonjet yeah.. I have no idea why.. I didn't even touch that file

**Comment by sinsinpub** [comment]
it failed on auto merging of this 'huge' reverse PR, duplicated occurred
https://github.com/KC3Kai/KC3Kai/commit/73c343017b165bba921506d18bb0324c19945de8#diff-2181f73accfb84cded588cf6ad5e6212

use `blame` view could be helped to find out which commit the change belongs to

if these duplicated codes/files are not supposed, overwrite them manually with current revisions of master may be a simple solution?

**Comment by Yukinyaa** [comment]
lol nailed it

File: src/library/modules/Kcsapi.js [hunk 1]
```diff
@@ -785,14 +785,7 @@ Previously known as "Reactor"
...
 			this["api_req_sortie/battle"].apply(this,arguments);
 		},
-		"api_req_sortie/ld_airbattle":function(params, response, headers){
-			KC3SortieManager.engageBattle(
-				response.api_data,
-				Math.floor((new Date(headers.Date)).getTime()/1000)
-			);
-			KC3Network.trigger("BattleStart");
-		},
-
```
Comment (Yukinyaa) [inline src/library/modules/Kcsapi.js]: ONLY this part is real

**Comment by Yukinyaa** [comment]
demonstration

the window looks like this.
it could be enhanced tho.

demonstration

**Comment by Yukinyaa** [comment]
yeah and could change gif to better ones
like these properly looping ones 

https://www.google.co.kr/url?sa=i&rct=j&q=&esrc=s&source=images&cd=&cad=rja&uact=8&ved=0ahUKEwiew9GL48XLAhVEx6YKHei7Cu4QjRwIBw&url=https%3A%2F%2Fwww.tumblr.com%2Ftagged%2Fkantai-collection-gif&psig=AFQjCNF0O9Jb8PE6GlWFm-BDAAlE8mSKgw&ust=1458236939147223


https://www.google.co.kr/url?sa=i&rct=j&q=&esrc=s&source=images&cd=&cad=rja&uact=8&ved=0ahUKEwjw3LvY48XLAhUB6aYKHRvJAFcQjRwIBw&url=http%3A%2F%2Fgiphy.com%2Fgifs%2Fkancolle-kantai-collection-shigure-l41lVCjI7RSx3PHAA&psig=AFQjCNEt2j6gkWxSfU-GDOcxAqBTxrTm4Q&ust=1458237372350270

**Comment by LostLogia4** [comment]
Loading from a clean save (no indexedDB entry) causes it to hang.
datasave

**Comment by Yukinyaa** [comment]
@LostLogia what blank entry are you talking about? After clicking clear record button on profile tab?

**Comment by LostLogia4** [comment]
Basically when I already loaded the game once but have **no record at all** in IndexedDB.

**Comment by Yukinyaa** [comment]
@LostLogia4 can you throw me what file you tried to restore? I had no problem loading same **almost blank** DB entey to kc3

**Comment by Yukinyaa** [comment]
fixed the code. you can try it with that code

**Pull request merged**